db/src/main/java/org/odk/collect/db/sqlite/SynchronizedDatabaseConnection.kt

```diff
@@ -0,0 +1,46 @@
+package org.odk.collect.db.sqlite
+
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import androidx.core.database.sqlite.transaction
+
+class SynchronizedDatabaseConnection(
+    context: Context,
+    path: String,
+    name: String,
+    migrator: DatabaseMigrator,
+    databaseVersion: Int
+) {
+    private val databaseConnection = DatabaseConnection(
```

Comment:
Hmm I'm wondering if haveing a separate class like `SynchronizedDatabaseConnection` makes sense at all, maybe we should just keep everything in `DatabaseConnection`?

Reply:
My thinking here is that "wrapping" the type rather than extending it (or expanding it) means that we ensure you can't call `writeableDatabase` or `readableDatabase` from `DatabaseEntitiesRepository`. One option would be to use extension but flip the hierarchy so that the stricter interface (`SynchronizedDatabaseConnection`) would be the super type of the more permissive one (`DatabaseConnection`). I think I prefer the wrapping though. What do you think?

Reply:
Ok makes sense.